File: app/src/main/java/com/adaldosso/spendykt/fragments/MonthlyListFragment.java
package com.adaldosso.spendykt.fragments;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.AbsListView;
import android.widget.TextView;

import com.adaldosso.spendykt.MainActivity;
import com.adaldosso.spendykt.R;
import com.adaldosso.spendykt.api.BaseExpense;
import com.adaldosso.spendykt.api.MonthlyExpense;
import com.adaldosso.spendykt.utils.BaseExpensesAdapter;
import com.adaldosso.spendykt.utils.MonthlyExpensesAdapter;
import com.adaldosso.spendykt.utils.SpendyUtils;

import java.util.List;

public class MonthlyListFragment extends SpendyListFragment implements AbsListView.OnItemClickListener {

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ((TextView) getActivity().findViewById(R.id.textView)).setText("MonthlyListFragment");
    }

    @Override
    protected int getLayout() {
        return R.layout.fragment_expenses_list;
    }

    @Override
    public void fillList() {
        SpendyUtils.getMonthlyExpenses(this::fillListCallback);
    }

    @Override
    protected BaseExpensesAdapter createExpensesAdapter(List<? extends BaseExpense> expenses, RecyclerView recyclerView) {
        return new MonthlyExpensesAdapter(expenses, recyclerView, getLayoutResourceId());
    }

    @Override
    protected int getLayoutResourceId() {
        return R.layout.monthly_expense;
    }

    @Override
    protected void onExpenseClick(BaseExpense baseExpense) {
        MonthlyExpense monthlyExpense = (MonthlyExpense) baseExpense;
        MainActivity activity = (MainActivity) getActivity();
        if (monthlyExpense.getYear() != null && monthlyExpense.getMonth() != null) {
            activity.loadExpenses(monthlyExpense.getYear(), monthlyExpense.getMonth());
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        fillList();
    }
}

File: app/src/main/java/com/adaldosso/spendykt/fragments/MonthlyListFragment.kt
package com.adaldosso.spendykt.fragments

import android.support.v7.widget.RecyclerView
import android.view.View.GONE
import android.view.View.VISIBLE
import android.widget.AdapterView
import android.widget.ProgressBar
import com.adaldosso.spendykt.MainActivity
import com.adaldosso.spendykt.R
import com.adaldosso.spendykt.api.BaseExpense
import com.adaldosso.spendykt.api.MonthlyExpense
import com.adaldosso.spendykt.utils.BaseExpensesAdapter
import com.adaldosso.spendykt.utils.MonthlyExpensesAdapter
import com.adaldosso.spendykt.utils.SpendyUtils.getMonthlyExpenses

class MonthlyListFragment : SpendyListFragment(), AdapterView.OnItemClickListener {

    override val layout: Int
        get() = R.layout.fragment_expenses_list

    override val layoutResourceId: Int
        get() = R.layout.monthly_expense

    override fun fillList() {
        showProgressBar()
        getMonthlyExpenses { fillListCallback(it) }
    }

    override fun hideProgressBar() {
        activity?.findViewById<ProgressBar>(R.id.progressBar)?.visibility = GONE
    }

    private fun showProgressBar() {
        activity?.findViewById<ProgressBar>(R.id.progressBar)?.visibility = VISIBLE
    }

    override fun createExpensesAdapter(expenses: List<BaseExpense>, recyclerView: RecyclerView?): BaseExpensesAdapter =
        MonthlyExpensesAdapter(expenses, recyclerView, layoutResourceId)

    override fun onExpenseClick(expense: BaseExpense) {
        val monthlyExpense = expense as MonthlyExpense
        val activity = activity as MainActivity
        if (monthlyExpense.getYear() != null && monthlyExpense.month != null) {
            activity.loadExpenses(monthlyExpense.getYear()!!, monthlyExpense.month!!)
        }
    }

    override fun onResume() {
        super.onResume()
        fillList()
    }
}
